merkle-tree/concurrent/src/test_utils.rs

```diff
@@ -0,0 +1,38 @@
+use light_hasher::Hasher;
+
+use crate::ConcurrentMerkleTree;
+
+pub fn assert_merkle_tree_initiallized<H, const HEIGHT: usize>(
+    merkle_tree: &ConcurrentMerkleTree<H, HEIGHT>,
+    height: usize,
+    changelog_capacity: usize,
+    roots_capacity: usize,
+    canopy_depth: usize,
+    expected_changelog_length: usize,
+    expected_roots_length: usize,
+    expected_next_index: usize,
+    expected_rightmost_leaf: &[u8; 32],
+) where
+    H: Hasher,
+{
+    assert_eq!(merkle_tree.height, height);
+    assert_eq!(merkle_tree.changelog_capacity, changelog_capacity);
+    assert_eq!(merkle_tree.changelog_length, expected_changelog_length);
+    assert_eq!(
+        merkle_tree.current_changelog_index,
+        expected_changelog_length.checked_sub(1).unwrap_or(0)
```

Comment:
Because index of the latest element is always equal to `length - 1`.

If you have any array `[0, 1, 2, 3, 4]`, the index of the latest element is `4` and the length is `5`.